### PR TITLE
qa/distros: add centos stream 9 as supported distro

### PR DESCRIPTION
**Pull request by cbodley**
<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: qa/distros/all/centos_9.stream.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+os_type: centos
+os_version: "9.stream"
+
+# enable the ceph/el9 copr for python packages that aren't in epel9 yet
+# see https://tracker.ceph.com/issues/58832
+overrides:
+  install:
+    ceph:
+      enable_coprs: [ceph/el9]
```
Comment (cbodley) [inline qa/distros/all/centos_9.stream.yaml:9]: note that this part depends on a teuthology PR https://github.com/ceph/teuthology/pull/1844

**Review by ktdreyer — APPROVED** [review]

**Comment by yuriw** [comment]
jenkins test make check

**Review by yuriw — APPROVED** [review]

**Comment by cbodley** [comment]
just pushed a rebase, but tests are currently failing with `Error: No matching repo to modify: powertools.` i believe that's due to a recent regression from https://github.com/ceph/ceph-cm-ansible/pull/742

**Comment by cbodley** [comment]
rebased again. it looks like my last rebase had accidentally dropped the `qa/distos: centos9 enables ceph/el9 copr` commit

**Comment by ljflores** [comment]
Rados approved, with the caveat that this is still causing a valgrind leak: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrellocomc6no1SSqS1762-wip-yuri2-testing-2023-07-17-0957-old-wip-yuri2-testing-2023-07-15-0802-old-wip-yuri2-testing-2023-07-13-1236-old-wip-yuri2-test

Valgrind leak is tracked in https://tracker.ceph.com/issues/61774

**Pull request merged**